### ISSUE TITLE
kernel BUG at fs/bcachefs/fsck.c:1194! [f2a03760652b]

### ISSUE THREAD

**Issue by FlyingWombat** [koverstreet/bcachefs#443]
This happened during fsck, after #442 ; I figured it should go in a separate issue, because earlier versions of bcachefs are affected.
Fsck fails every time now, with message "kernel BUG at fs/bcachefs/fsck.c:1194!"
Same thing happens on either of the two versions listed below.

debug build dmesg log:
[bcachefs-5.19.0-gf2a03760652b-fsck-fail-debug.txt](https://github.com/koverstreet/bcachefs/files/9275629/bcachefs-5.19.0-gf2a03760652b-fsck-fail-debug.txt)

**Version**
Linux v5.19 bcachefs f2a03760652bfb0b1a33a52e25f5d96d570b3da2
also affects v5.18.16 bcachefs 5549e504645ac9ae570323eab6417b560a7def9b

**Generic info**
superblock: 
[bcachefs-superblock_2022-08-06.txt](https://github.com/koverstreet/bcachefs/files/9275640/bcachefs-superblock_2022-08-06.txt)
12th-gen Intel CPU ; ibt=off

built kernel with debug flags:
```
CONFIG_PREEMPT=y
CONFIG_BCACHEFS_DEBUG=y
CONFIG_KALLSYMS=y
CONFIG_KALLSYMS_ALL=y
CONFIG_DEBUG_FS=y
CONFIG_DYNAMIC_FTRACE=y
CONFIG_FTRACE=y
```

**Comment by FlyingWombat** [comment]
Seems to be fixed as of c849702be7ec516fb28980e6c2f20d0b54f4e0dc
